src/pstore/pstore.c

```diff
@@ -141,12 +139,14 @@ static int move_file(PStoreEntry *pe, const char *subdir) {
         field = strjoina("MESSAGE=PStore ", pe->dirent.d_name, suffix);
         iovec[n_iovec++] = IOVEC_MAKE_STRING(field);
 
-        field_size = strlen("FILE=") + pe->content_size;
-        field = malloc(field_size);
```

Comment:
Currently the code seems to be based on the assumption that `malloc` is always called here. If it isn't the memory allocated via `strjoina` `field` points to is freed because of `_cleanup_free_` causing `systemd-pstore` to crash with:
```
==8432==ERROR: AddressSanitizer: attempting free on address which was not malloc()-ed: 0x7ffc6ff9de60 in thread T0
    #0 0x7fe77d3f0840 in __interceptor_free (/lib64/libasan.so.5+0xef840)
    #1 0x404235 in freep ../src/basic/alloc-util.h:83
    #2 0x404235 in move_file ../src/pstore/pstore.c:120
    #3 0x406554 in run ../src/pstore/pstore.c:385
    #4 0x406554 in main ../src/pstore/pstore.c:390
    #5 0x7fe77b29e412 in __libc_start_main (/lib64/libc.so.6+0x24412)
    #6 0x4024ad in _start (/home/vagrant/systemd/build/systemd-pstore+0x4024ad)

Address 0x7ffc6ff9de60 is located in stack of thread T0
SUMMARY: AddressSanitizer: bad-free (/lib64/libasan.so.5+0xef840) in __interceptor_free
==8432==ABORTING
```

I think it would be better to fix https://github.com/systemd/systemd/issues/13122 by replacing `stpcpy` (which tries to write a null byte `field` doesn't have room for) with `memcpy`. I'd also avoid `strjoina` there in general or add another variable declared without `_cleanup_free_`.

Reply:
Ouch. Will fix that.